src/prefs/TelemetryPrefs.cpp

```diff
@@ -0,0 +1,113 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  TelemetryPrefs.cpp
+
+  Dmitry Vedenko
+
+
+*******************************************************************//**
+
+\class TelemetryPrefs
+\brief A PrefsPanel to enable/disable user data sharing.
+
+*//*******************************************************************/
+
+#include "../Audacity.h"
+#include "TelemetryPrefs.h"
+
+#include <wx/defs.h>
+#include <wx/hyperlink.h>
+
+#include "../ShuttleGui.h"
+
+////////////////////////////////////////////////////////////////////////////////
+
+TelemetryPrefs::TelemetryPrefs(wxWindow * parent, wxWindowID winid)
+:  PrefsPanel(parent, winid, XO("Analytics"))
+{
+   Populate();
+}
+
+TelemetryPrefs::~TelemetryPrefs()
+{
+}
+
+ComponentInterfaceSymbol TelemetryPrefs::GetSymbol()
+{
+   return TELEMETRY_PREFS_PLUGIN_SYMBOL;
+}
+
+TranslatableString TelemetryPrefs::GetDescription()
+{
+   return XO("Preferences for Analytics");
+}
+
+wxString TelemetryPrefs::HelpPageName()
+{
+   return "Analytics_Preferences";
+}
+
+void TelemetryPrefs::Populate()
+{
+   //------------------------- Main section --------------------
+   // Now construct the GUI itself.
+   // Use 'eIsCreatingFromPrefs' so that the GUI is
+   // initialized with values from gPrefs.
+   ShuttleGui S(this, eIsCreatingFromPrefs);
+   PopulateOrExchange(S);
+   // ----------------------- End of main section --------------
+}
+
+void TelemetryPrefs::PopulateOrExchange(ShuttleGui & S)
+{
+   S.SetBorder(2);
+
+   S.StartStatic(XO("Analytics"));
+   {
+       S.StartHorizontalLay (wxEXPAND);
+       {
+           S.StartVerticalLay (0);
+           {
+               S.TieCheckBox ({}, { wxT ("/Telemetry/Enabled"), false });
+           }
+           S.EndVerticalLay ();
+
+           S.StartVerticalLay ();
+           {
+               S.AddFixedText (XO ("Share anonymous analytics and crash reports"));
+               S.AddFixedText (XO ("Help Audacity improve its service, by allowing analytics of usage and data."));
```

Comment:
This is too vague. Tell users what is being collected and why.

Reply:
"Anonymous" is unlikely to be correct. Audacity cannot guarantee that Google or Yandex won't be able to link the user's IP address to their identity, so you'd better avoid making promises which might be seen as lies.

Reply:
Also, if the UUID is saved it is at best *pseudo*nymous.

Reply:
However, if Google or Yandex says it's anonymous then the blame shifts to these corporations in case it turns out otherwise, which in this case saying that it's "Anonymous" is correct.

Reply:
> However, if Google or Yandex says it's anonymous then the blame shifts to these corporations in case it turns out otherwise, which in this case saying that it's "Anonymous" is correct.

That's not how law usually works. And we already *know* that it's not anonymous. Google is an advertising company.